### PR TITLE
feat: react helpers

### PR DESCRIPTION
**Pull request by nicklasl**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing
- [ ] Relevant documentation updated
- [ ] linter/style run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes
- [ ] Tested in a corresponding example app

### CODE REVIEW

File: packages/react-helpers/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Confidence React helpers
+
+![](https://img.shields.io/badge/lifecycle-beta-a0c3d2.svg)
+
+This is a client for direct access to the Confidence API, it is not intended for end user use, it is an internal package
+used in `@spotify-confidence/openfeature-web-provider` and `@spotify-confidence/openfeature-server-provider` packages.
```
Comment (nicklasl) [inline packages/react-helpers/README.md:6]: Needs update.

File: packages/react-helpers/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "@spotify-confidence/react-helpers",
```
Comment (nicklasl) [inline packages/react-helpers/package.json:2]: update name?

File: packages/react-helpers/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "scripts": {
+    "bundle": "rollup -c && api-extractor run",
+    "build": "tsc"
+  },
+  "engineStrict": true,
+  "engines": {
+    "node": ">=18.17.0"
+  },
+  "publishConfig": {
+    "access": "public",
```
Comment (nicklasl) [inline packages/react-helpers/package.json:18]:
```suggestion
    "registry": "https://registry.npmjs.org/",
    "access": "public",
```

File: packages/react-helpers/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+  getFlag<T extends Value>(path: string, defaultValue: T): FlagEvaluation<T> {
+    const evaluation = super.getFlag(path, defaultValue);
+    if (evaluation.reason === 'ERROR' && 'then' in evaluation && isRendering()) throw evaluation;
+    return evaluation;
+  }
+
+  setContext(context: Context): boolean {
+    if (super.setContext(context)) {
+      const promise = this.resolveFlags();
```
Comment (nicklasl) [inline packages/react-helpers/src/index.tsx:52]: great work being able to trigger the resolve from this integration 👍

File: packages/sdk/src/Confidence.ts [hunk 5]
```diff
@@ -247,13 +259,16 @@ export class Confidence implements EventSender, Trackable, FlagResolver {
...
-    const flagResolverClient = new FlagResolverClient({
+    let flagResolverClient: FlagResolverClient = new FetchingFlagResolverClient({
       clientSecret,
       fetchImplementation,
       sdk,
       environment,
       region,
     });
+    if (environment === 'client') {
+      flagResolverClient = new CachingFlagResolverClient(flagResolverClient, 3000_000);
```
Comment (nicklasl) [inline packages/sdk/src/Confidence.ts:270]: I think there's one or two 0s too much here? 3 thousand seconds?
Also, did we decide to extract this to be a setting? (can defer to a next release)

File: packages/react-helpers/package.json (hunk 3 above)
Comment (andreas-karlsson) [reply packages/react-helpers/package.json:18]: Fix by constraints

**Review by andreas-karlsson — APPROVED** [review]

**Pull request merged**